New package: ComponentLogging v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: ComponentLogging
- Repository: https://github.com/abcdvvvv/ComponentLogging.jl
- Created by: @abcdvvvv
- Version: v0.1.0
- Commit: 7146a792fafc41c2220c247c759507abf76d79b0
- Reviewed by: @abcdvvvv
- Reference: https://github.com/abcdvvvv/ComponentLogging.jl/commit/7146a792fafc41c2220c247c759507abf76d79b0#commitcomment-166849261
- Release notes:
<!-- BEGIN RELEASE NOTES -->
`````
- feat: Include module name in PlainLogger output. Add module name to the log output format in PlainLogger, displaying it before the filename for better context in log messages.
`````
<!-- END RELEASE NOTES -->

<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd64195a8368320cc6c979146d5182b43c70af59f42dfe164773528a69898d30a9766b4c36e06b828716d0cf502913fed369e4344dcba3e35a422a6797f79507b0ef7f0bd0880b21b7bb5a0373f44dbe13156d0e37dab73bb4b2f6439516e76a47c8890d58a328d048dbb0d59f429f2bbc1f927f691b62b65b7c7e03cd602d3a66e4dfd08c72f35f54b307844f8163da326d13dd2bb24b83ad45ef46455614210744c0115a4070c39e8f2d6a08b562d73a3 -->

goerz commented:
Thank you for submitting your package! Could you add a little bit of documentation before registering? [At the very least](https://discourse.julialang.org/t/when-does-it-make-sense-to-post-a-new-package-to-the-julia-repository/108701/4), that would be a description of the package's purpose and a small usage example in the README. An important part of packages in General is that any potential user can figure out what the package is about and how to get started with using it. That is really difficult when there is no documentation.

In general, I definitely recommend setting up a [Documenter](https://documenter.juliadocs.org/stable/)-based documentation. In your case, you already have the documentation badges in the README, but these currently aren't working, as your documentation builds are failing in CI. I think you might be able to fix that failure by adding a [`[sources]` section](https://pkgdocs.julialang.org/dev/toml-files/#The-%5Bsources%5D-section) to [`docs/Project.toml`](https://github.com/abcdvvvv/ComponentLogging.jl/blob/master/docs/Project.toml):

```
[sources]
ComponentLogging = {path = ".."}
```

This requires using at Julia 1.11 to build the documentation, so you should [change that in your `.github/workflows/Docs.yml` file](https://github.com/abcdvvvv/ComponentLogging.jl/blob/cc37c8846ff6fea715cd9f53e4f99010f9c6492e/.github/workflows/Docs.yml#L18) (probably best to just set it to `'1'`).

Lastly: Could this package be hosted under the [JuliaLogging organization](https://github.com/JuliaLogging)? By putting packages in a well-established organization with multiple owners, it avoids the possibility of becoming permanently unmaintained.

goerz commented:
> Give me a little time! I'll definitely add it as soon as possible!

No rush! There's the three-day waiting period, in any case. You can update the registration at any point just by retriggering it (do not change the version number). This will not reset the waiting period.

> Yes, I agree to place it in the JuliaLogging organization. Is there anything I need to do regarding this?

Let me tag @c42f, @fredrikekre and @ericphanson as members of the org, who can probably help you with the move or point you to an org owner who can.

The org owner may have their own guidelines for what they would consider requirements for adding the package. My recommendation would be to resolve the same documentation issues I already raised with respect to the registration in general:

* In the README, explain what problem the package solves and how it fits in with other existing packages in the JuliaLogging org
* Make sure that there is sufficient API documentation

[noblock]

c42f commented:
Interesting. So this package makes the decision that loggers belong to `Module`s, whereas Base logging thinks that loggers belong to `Task`s (ie, are not inherently global)? From a quick look, that's the core semantic difference which affects both performance and behavior in concurrent code. It's quite a significant difference so you might mention it in the README.

I'm fine with this going in JuliaLogging if you want.

[noblock]

abcdvvvv commented:
[noblock]

JuliaRegistrator committed:
New package: ComponentLogging v0.1.0

UUID: 3f6ec708-df3b-46b4-8b70-ec9fa53b9f7d
Repo: https://github.com/abcdvvvv/ComponentLogging.jl.git
Tree: 87c8f7ecc91963d6fc4bd44246659ee63d3cce52

Registrator tree SHA: 50f504d641745716a5b3eabaf681d3a4937d2ae3

c42f commented:
@abcdvvvv The transfer expired, sorry!! Try again?

[noblock]

c42f commented:
See https://github.com/JuliaLogging/ComponentLogging.jl

@abcdvvvv I've invited you as a member of JuliaLogging so you can be added as admin of the repository.

[noblock]

abcdvvvv commented:
I see, thank you for your help! :)

[noblock]

c42f commented:
Ok you should be all set up to admin the repo in the new location now :)

[noblock]

goerz commented:
> By the way, do I still have to wait for this PR to be merged before I can find my package in the Julia registry?

No, you trigger a new registration from the new location. This will create a new PR, and I'll close this one in favor of the new one.

c42f commented:
Yes, this needs to be merged.

Ideally the registration PR would be updated with the new URL for the project. I guess you can do that by re-triggering the registration (actually not sure, I don't do transfers often)

[noblock]

goerz commented:
Make sure to adjust any references to the old location first, though. For example, the documentation badges in the README still reference the old https://abcdvvvv.github.io/ComponentLogging.jl/stable/

You may also have to to re-enable GitHub pages and make sure CI / deploying the documentation works as expected.

abcdvvvv commented:
> No, you trigger a new registration from the new location. This will create a new PR, and I'll close this one in favor of the new one.

Okay, I understand. Thanks for your guide.

I will follow that.

> Make sure to adjust any references to the old location first

I see, I will fix the doc. and CI before making a new PR.

abcdvvvv commented:
Hello, @goerz.  For your information, the package has been re-registered as v0.1.1 (see: https://github.com/JuliaRegistries/General/pull/139833), so this v0.1.0 registration is no longer needed.

Please feel free to close this PR. Thank you!
